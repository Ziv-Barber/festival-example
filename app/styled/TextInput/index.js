// @flow

import React from 'react' // eslint-disable-line no-unused-vars
import styled from 'styled-components'

// Other components that we need:

// Internationalization (i18n):
// import { FormattedMessage } from 'react-intl'
// import createMessages from './messages'

// Material-ui stuff:
import TextField from '@material-ui/core/TextField'

// Our types:
import type { TextInputPropsType } from './index.types'

/**
 * TextInput react component.
 * @reactProps {string} text Parameters for the component.
 * @return New element.
 */
const TextInput = (props: TextInputPropsType) => (
  <TextField {...props} variant="outlined" margin="normal" />
)

export default styled(TextInput)`
  color: #fff;
`

// In case that someone need our types:
export type { TextInputPropsType as Props }
