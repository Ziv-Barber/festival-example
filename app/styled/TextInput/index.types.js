// @flow

import * as React from 'react' // eslint-disable-line no-unused-vars

export type TextInputPropsType = {
  /** The class to use for the root element. */
  className?: string,

  /** Children to put inside this component. */
  children?: React.Node,

  /**
   * This property helps users to fill forms faster, especially on mobile devices.
   * The name can be confusing, as it's more like an autofill.
   * You can learn more about it here:
   * https://html.spec.whatwg.org/multipage/form-control-infrastructure.html#autofill
   */
  autoComplete?: string,

  /**
   * If `true`, the input will be focused during the first mount.
   */
  autoFocus?: boolean,

  /**
   * The default value of the `Input` element.
   */
  defaultValue?: string | number,

  /**
   * If `true`, the input will be disabled.
   */
  disabled?: boolean,

  /**
   * If `true`, the label will be displayed in an error state.
   */
  error?: boolean,

  /**
   * Properties applied to the [`FormHelperText`](/api/form-helper-text/) element.
   */
  FormHelperTextProps?: Object,

  /**
   * If `true`, the input will take up the full width of its container.
   */
  fullWidth?: boolean,

  /**
   * The helper text content.
   */
  helperText?: React.Node,

  /**
   * The id of the `input` element.
   * Use this property to make `label` and `helperText` accessible for screen readers.
   */
  id?: string,

  /**
   * Properties applied to the [`InputLabel`](/api/input-label/) element.
   */
  InputLabelProps?: Object,

  /**
   * Properties applied to the `Input` element.
   */
  InputProps?: Object,

  /**
   * Attributes applied to the native `input` element.
   */
  inputProps?: Object,

  /**
   * Use this property to pass a ref callback to the native input component.
   */
  inputRef?: Function | Object,

  /**
   * The label content.
   */
  label?: React.Node,

  /**
   * If `dense` or `normal`, will adjust vertical spacing of this and contained components.
   */
  margin?: 'none' | 'dense' | 'normal',

  /**
   * If `true`, a textarea element will be rendered instead of an input.
   */
  multiline?: boolean,

  /**
   * Name attribute of the `input` element.
   */
  name?: string,

  /**
   * @ignore
   */
  onBlur?: Function,

  /**
   * Callback fired when the value is changed.
   *
   * @param {object} event The event source of the callback.
   * You can pull out the new value by accessing `event.target.value`.
   */
  onChange?: Function,

  /**
   * @ignore
   */
  onFocus?: Function,

  /**
   * The short hint displayed in the input before the user enters a value.
   */
  placeholder?: string,

  /**
   * If `true`, the label is displayed as required and the input will be required.
   */
  required?: boolean,

  /**
   * Number of rows to display when multiline option is set to true.
   */
  rows?: string | number,

  /**
   * Maximum number of rows to display when multiline option is set to true.
   */
  rowsMax?: string | number,

  /**
   * Render a `Select` element while passing the `Input` element to `Select` as `input` parameter.
   * If this option is set you must pass the options of the select as children.
   */
  select?: boolean,

  /**
   * Properties applied to the [`Select`](/api/select/) element.
   */
  SelectProps?: Object,

  /**
   * Type attribute of the `Input` element. It should be a valid HTML5 input type.
   */
  type?: string,

  /**
   * The value of the `Input` element, required for a controlled component.
   */
  value?: string | number | boolean | Array<string | number | boolean>,

  /**
   * The variant to use.
   */
  variant?: 'standard' | 'outlined' | 'filled'
}
