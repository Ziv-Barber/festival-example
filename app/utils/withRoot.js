// @flow

import * as React from 'react' // eslint-disable-line no-unused-vars
import { MuiThemeProvider } from '@material-ui/core/styles'
import { ThemeProvider } from 'styled-components'
import CssBaseline from '@material-ui/core/CssBaseline'
import theme from '../theme'

/**
 * withRoot HOC to wrap a react component with the app's theme provider.
 */
const withRoot = (Component: React.ComponentType<any>) => {
  const withRootBase = (props: any) => (
    <MuiThemeProvider theme={theme}>
      <ThemeProvider theme={theme}>
        <React.Fragment>
          <CssBaseline />
          <Component {...props} />
        </React.Fragment>
      </ThemeProvider>
    </MuiThemeProvider>
  )

  // $FlowFixMe
  withRootBase.displayName = `withRoot(${Component.displayName ||
    Component.name})`

  return withRootBase
}

export default withRoot
