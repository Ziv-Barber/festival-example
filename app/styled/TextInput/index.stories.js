import React from 'react' // eslint-disable-line no-unused-vars
import { storiesOf } from '@storybook/react'
import TextInput from './index'

storiesOf('TextInput', module).add(
  'basic',
  () => <TextInput text="Basic example" />,
  { notes: { markdown: 'This is an example to the TextInput component.' } }
)
