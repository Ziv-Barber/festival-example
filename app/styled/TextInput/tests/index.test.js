// @flow

import React from 'react' // eslint-disable-line no-unused-vars
import { render, cleanup } from 'react-testing-library'

import TextInput from '../index'
import withRoot from '../../../utils/withRoot'

// For flow:
let { describe, it, afterEach, expect } = global

const CompToTest = () => (
  <TextInput id="test_field" label="Name" value="This is a test" />
)

/** @test {TextInput} */
describe('<TextInput />', () => {
  afterEach(cleanup)

  /** @test {TextInput} */
  it('should match snapshort', () => {
    const App = withRoot(CompToTest)
    const { container } = render(<App />)
    expect(container).toMatchSnapshot()
  })

  /** @test {TextInput} */
  it('should render', () => {
    const App = withRoot(CompToTest)
    render(<App />)
    // const { getByText } = render(<App />)
    // expect(getByText(/^This is a text/)).not.toBeNull()
  })
})
