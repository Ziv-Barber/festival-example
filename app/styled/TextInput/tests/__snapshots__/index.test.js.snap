// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<TextInput /> should match snapshort 1`] = `
<div>
  <div
    class="MuiFormControl-root-1 MuiFormControl-marginNormal-2 TextInput-k6q9l7-0 jjzSuf"
  >
    <label
      class="MuiFormLabel-root-16 MuiFormLabel-filled-20 MuiInputLabel-root-5 MuiInputLabel-formControl-10 MuiInputLabel-animated-13 MuiInputLabel-shrink-12 MuiInputLabel-outlined-15"
      data-shrink="true"
      for="test_field"
    >
      Name
    </label>
    <div
      class="MuiInputBase-root-36 MuiOutlinedInput-root-23 MuiInputBase-formControl-37"
    >
      <fieldset
        aria-hidden="true"
        class="MuiPrivateNotchedOutline-root-53 MuiOutlinedInput-notchedOutline-30"
        style="padding-left: 8px;"
      >
        <legend
          class="MuiPrivateNotchedOutline-legend-54"
          style="width: 0px;"
        >
          <span>
            â€‹
          </span>
        </legend>
      </fieldset>
      <input
        aria-invalid="false"
        class="MuiInputBase-input-46 MuiOutlinedInput-input-31"
        id="test_field"
        type="text"
        value="This is a test"
      />
    </div>
  </div>
</div>
`;
